index.md

```diff
@@ -8,6 +8,573 @@ This is some welcome text.
 
 This is some helpful description of the C0111 error.
 
+
+
+
+### R0201: No self use {#R0201}
+
+If a method (a function in a class) does not make use of the 'self' (or
+first) argument, that means the function is not performing anything that
+is related to the object itself. This means the function could be moved
+outside of the class since none of the code inside makes use of anything
+inside the class it's defined.
+
+```python
+class NoSelfUsage:
+    def __init__(self):
+        self.a = 42
+
+    def no_self(self, num):
+        num = num + 2
+        print(num)
+
+# You would fix it as follows by moving it outside the class:
+def no_self(num):
+	num = num + 2
+    print(num)
+```
+
+
+
+### E0202: Method hidden {#E0202}
+
+If you accidentally mask a method with a field, it can cause other code to
+attempt to invoke what it believes to be a method, which will fail since
+it has become a field instead (a variable). The syntax will error out, and
+this can be particularly dangerous when masking the function when a parent
+class relies on calling the method.
+
+```python
+class Example(object):
+    def field(self, num):
+        return num
+    def __init__(self):
+        self.field = 'Masking the function with this string'
+
+# If you call Example().field(num), it will error out since we masked it
+```
+
+
+
+### E0203: Access to member before definition {#E0203}
+
+Before trying to use a member of a class, it should have been defined at
+some point. If you try to use it before assigning to it, Python cannot
+resolve the value and an error will occur.
+
+```python
+class MyClass:
+	def __init__(self):
```

Comment:
Some good stuff here 👍, although [don't shoot the messenger] it looks like spaces (above here) are mixed with tabs (generally below here) 👇